packages/cli/src/lib/packager/createDistWorkspace.ts

```diff
@@ -50,6 +51,19 @@ const UNSAFE_PACKAGES = [
   ...Object.keys(cliDevDependencies),
 ];
 
+const ROOT_DIR = resolvePath('..', '..');
+const NPM_LOCKFILE = 'package-lock.json';
+const YARN_LOCKFILE = 'yarn.lock';
+const useNpm =
+  !existsSync(joinPath(ROOT_DIR, YARN_LOCKFILE)) &&
+  existsSync(joinPath(ROOT_DIR, NPM_LOCKFILE));
+const LOCKFILE = useNpm ? NPM_LOCKFILE : YARN_LOCKFILE;
+const PKG_COMMAND = useNpm ? 'npm' : 'yarn';
+if (useNpm)
+  console.log(
+    `Using npm because of ${NPM_LOCKFILE} and no ${YARN_LOCKFILE} in ${ROOT_DIR}`,
+  );
+
```

Comment:
Given this request - https://github.com/backstage/backstage/issues/20658 - to support `pnpm` would it make sense to make this more "open" to other lock file formats?

Reply:
Probably not hard to add a further case for pnpm or to generalize with a map object.

But I won’t have time to test with pnpm in addition. I know this version works as I have a local patch running in the CI so I’d like to keep it as is for now but if someone wants to build on top of it of course that would be great.

Reply:
@dbrugger et al. We have a bare-bones Backstage implementation in a monorepo where the root is using `pnpm`, but the Backstage project is using `yarn`. If we could somehow apply the changes in this PR, we could test it against this repo:
https://github.com/ourchitecture/devex/tree/main/src/backstage/ourstage

Reply:
See https://github.com/backstage/backstage/pull/20670#discussion_r1369750361